### ISSUE TITLE
FeatureScopeSessionWithContext.isVisible flags stuff as invisible that is visible in java 

### ISSUE THREAD

**Issue by cdietrich** [eclipse/xtext-extras#229]
the following works in java

```
package xxx;

public class Base {
	
	protected String field;

}
```

```
package yyy;

import xxx.Base;

public class Sub extends Base{

    

    public static class Inner {

        public void doSthWith(Sub sub) {

            sub.field = "Hello";

        }

    }

}


```

but when replacing sub with its Xtend version it no longer works

```
package yyy

import xxx.Base

class Sub extends Base {
    static class Inner {
        def void doSthWith(Sub sub) {
            sub.field = "Hello"
        }
    }
}
```
The error message is: The field field is not visible

The invisibility is calculated by FeatureScopeSessionWithContext.isVisible

I Thinlk it should not only do a `receiverType.isSubtypeOf(contextType)`
but a `receiverType.isSubtypeOf(outerType)` for all outer types

**Comment by cdietrich** [comment]
https://docs.oracle.com/javase/specs/jls/se9/html/jls-6.html#jls-6.6.2.1

**Comment by cdietrich** [comment]
possible solution
```
 if (receiverType != null) {
				if (receiverType.isSubtypeOf(contextType)) {
					return true;
				}
				EObject container = contextType.eContainer();
				while (container instanceof JvmType) {
					if (receiverType.isSubtypeOf((JvmType)container)) {
						return true;
					}
					container = container.eContainer();
				}
			}

```

**Comment by cdietrich** [comment]
Fixed in 2.14
